client/src/context/useBookingContext.tsx

```diff
@@ -0,0 +1,37 @@
+import { useContext, createContext, FunctionComponent, useState, useEffect } from 'react';
+import { Booking } from '../interface/Booking';
+import { fetchBookings, updateBooking } from '../helpers/APICalls/booking';
+
+interface IBookingContext {
+  bookings: Booking[];
+  updateBookings: (bookingStatus: string, requestId: string) => void;
+}
+
+export const BookingContext = createContext<IBookingContext>({
+  bookings: [],
+  updateBookings: () => null,
+});
+
+export const BookingProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [bookings, setBookings] = useState<Booking[]>([]);
+
+  useEffect(() => {
+    const checkFetchBookings = async () => {
+      await fetchBookings().then((data) => {
```

Comment:
If there is an error fetching the bookings, why don't we consider updating the snack bar here? :)